addon/ng2/utilities/get-dependent-files.ts

```diff
@@ -0,0 +1,132 @@
+'use strict';
+
+import * as fs from 'fs';
+import * as ts from 'typescript';
+import * as glob from 'glob';
+import * as path from 'path';
+import * as denodeify from 'denodeify';
+
+/**
+ * Declare an Interface where key is a string and value is either a string or a number.
+ * Use for storing a string literal, start position and end posittion of ImportClause node kinds.
+ */
+export interface ModuleImport {
+  specifierText: string;
+  pos: number;
+  end: number;
+};
+
+export interface ModuleMap {
+  [key: string]: ModuleImport[];
+}
+
+/**
+ * Create a SourceFile as defined by Typescript Compiler API.
+ * Generate a AST structure from a source file.
+ * 
+ * @param fileName source file for which AST is to be extracted
+ */
+export function createTsSourceFile(fileName: string): Promise<ts.SourceFile> {
+  const readFile = denodeify(fs.readFile);
+  return readFile(fileName, 'utf8')
+    .then((contents: string) => {
+      return ts.createSourceFile(fileName, contents, ts.ScriptTarget.ES6, true);
+    });
+}
+
+/**
+ * @return traverses through ts.Node and returns an array of moduleSpecifiers
+ * 
+ * @param {ts.SourceFile} node: Typescript Node of whose AST is being traversed
+ * 
+ */
+export function getImportClauses(node: ts.SourceFile): ModuleImport[] {
+  return node.statements
+    .filter(node => node.kind === ts.SyntaxKind.ImportDeclaration)  // Only Imports.
+    .map((node: ts.ImportDeclaration) => {
+      let moduleSpecifier = node.moduleSpecifier;
+      return {
+        specifierText: moduleSpecifier.getText().slice(1, -1),
+        pos: moduleSpecifier.pos,
+        end: moduleSpecifier.end
+      };
+    });
+}
+
+/** 
+ * @return a boolean value after it searches for a barrel (index.ts by convention) in a given path
+ * 
+ * @param filePath
+ * @param isDir boolean value to declare whether filePath is a directory or not. 
+ */
+export function hasIndexFile(filePath: string, isDir: boolean): Boolean {
+  let indexFile: string[];
+  if (isDir) {
+    indexFile = glob.sync(path.join(filePath, 'index.ts'), { nodir: true });
+  } else {
+    let pathDir = path.dirname(filePath);
+    indexFile = glob.sync(path.join(pathDir, 'index.ts'), { nodir: true });
+  };
+  return (indexFile.length > 0) ? true : false;
+}
+
+
+/**
+ * Return a map of all dependent file/s' path with their moduleSpecifier object
+ * (text, pos, end)
+ * 
+ * @param fileName file upon which other files depend 
+ * @param rootPath root of the project
+ * 
+ */
+export function getDependentFiles(fileName: string, rootPath: string): Promise<ModuleMap> {
+  const files = glob.sync(path.join(rootPath, '**/*.*.ts'), { nodir: true });
+
+  let allPromiseFiles = files
+    .map(file => createTsSourceFile(file))
+    .map(tsFile => tsFile
+      .then((node: ts.SourceFile) => getImportClauses(node)));
```

Comment:
This is hard to follow because `tsFile` is actually misnamed; it's a promise of a `SourceFile`, really. It's kinda confusing to have a map return a promise and another map wait on that promise. I feel it's more natural to do it the other way.

What you could also do it reverse the logic a little bit:

```typescript
return Promise.all(files.map(file => createTsSourceFile(file)))
  .then(tsFiles => tsFiles.map(file => getImportClauses(file))
  .then((moduleSpecifiers: ModuleImport[][]) => { ... })
  ...
```

So waiting on the createTsSourceFile instead of the getImportClauses return. It's up to you; what do you think?